solver/result/attestation.go

```diff
@@ -8,8 +8,9 @@ import (
 type Attestation struct {
 	Kind pb.AttestationKind
 
-	Ref  string
-	Path string
+	Ref     string
+	Path    string
+	Content []byte
```

Comment:
> Also, I'm not sure why you don't need conditions

Woops. It worked before because I was creating the `Content` attestations in `Unbundle` after this check. I've updated the checks here, and also made sure to add an explicit error in the gateway, if trying to send `Content` over it. Not sure about this design, depends on whether we use `Callback` instead of `Content`.

Reply:
Have updated with `Callback` from https://github.com/moby/buildkit/pull/3240 - some minor differences in the conditions, for counting the number of attestations, etc, but should be functionally equivalent.